x-pack/plugins/lists/public/exceptions/hooks/persist_exception_item.test.ts

```diff
@@ -45,51 +46,77 @@ describe('usePersistExceptionItem', () => {
   });
 
   test('"isLoading" is "true" when exception item is being saved', async () => {
-    await act(async () => {
-      const { result, rerender, waitForNextUpdate } = renderHook<
-        PersistHookProps,
-        ReturnPersistExceptionItem
-      >(() => usePersistExceptionItem({ http: mockKibanaHttpService, onError }));
+    const defaultImplementation = (api.addExceptionListItem as jest.Mock).getMockImplementation();
+
+    (api.addExceptionListItem as jest.Mock).mockImplementation(function () {
+      // delay response to ensure that isLoading is set to true for a while
+      return new Promise((resolve) =>
+        setTimeout(() => resolve(getExceptionListItemSchemaMock()), 1000)
+      );
+    });
+
+    // setup fake timers before rendering
+    jest.useFakeTimers();
+
+    const { result } = renderHook<PersistHookProps, ReturnPersistExceptionItem>(() =>
+      usePersistExceptionItem({ http: mockKibanaHttpService, onError })
+    );
+
+    await waitFor(() =>
+      expect(result.current).toEqual([{ isLoading: false, isSaved: false }, result.current[1]])
+    );
 
-      await waitForNextUpdate();
+    await act(async () => {
       result.current[1](getCreateExceptionListItemSchemaMock());
-      rerender();
+    });
 
+    await waitFor(() => {
+      expect(api.addExceptionListItem).toHaveBeenCalled();
       expect(result.current).toEqual([{ isLoading: true, isSaved: false }, result.current[1]]);
     });
+
+    await jest.advanceTimersByTimeAsync(500);
+
+    await waitFor(() => {
+      expect(result.current).toEqual([{ isLoading: false, isSaved: true }, result.current[1]]);
+    });
+
+    // restore real timers
+    jest.useRealTimers();
```

Comment:
`jest.useRealTimers();` should be put in `afterEach` hook to prevent possible effect on other tests if this one fails

Reply:
Done ✔️